Source/Scene/SkyAtmosphere.js

```diff
@@ -169,11 +211,22 @@ define([
                 defines : ['SKY_FROM_SPACE'],
                 sources : [SkyAtmosphereVS]
             });
+
+            var fsColorCorrect = new ShaderSource({
+                defines : ['COLOR_CORRECT'],
+                sources : [SkyAtmosphereFS]
+            });
+
             this._spSkyFromSpace = ShaderProgram.fromCache({
                 context : context,
                 vertexShaderSource : vs,
                 fragmentShaderSource : SkyAtmosphereFS
             });
+            this._spSkyFromSpaceColorCorrect = ShaderProgram.fromCache({
```

Comment:
The color correct shaders could just be compiled on demand. Compile them when h,s,b are not 0.0 and the shaders aren't yet defined.